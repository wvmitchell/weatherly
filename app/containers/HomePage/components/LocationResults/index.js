/**
*
* LocationResults
*
*/

import React from 'react';
import {
  RaisedButton,
} from 'material-ui';


class LocationResults extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <div>
        <h2>We found a few matches, is one of these what you are looking for?</h2>
        <div className="location-options">
          { this.props.results.map((result) => (
            <RaisedButton
              backgroundColor="primary"
              className="location-option"
              onClick={() => this.props.selectSuggestion(result.l)}
              key={result.l}
            >
              {`${result.city} ${result.state}, ${result.country_name}`}
            </RaisedButton>)
            )
          }
        </div>
      </div>
    );
  }
}

LocationResults.propTypes = {
  results: React.PropTypes.array,
  selectSuggestion: React.PropTypes.func,
};

export default LocationResults;
