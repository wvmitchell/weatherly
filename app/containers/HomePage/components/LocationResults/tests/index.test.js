import React from 'react';
import { shallow } from 'enzyme';

import LocationResults from '../index';

describe('<LocationResults />', () => {
  it('matches the snapshot', () => {
    const renderedComponent = shallow(
      <LocationResults
        results={[]}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
