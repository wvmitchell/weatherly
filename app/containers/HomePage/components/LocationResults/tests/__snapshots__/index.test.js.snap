exports[`<LocationResults /> matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        We found a few matches, is one of these what you are looking for?
    </h2>
    <div
        className="location-options">
        
    </div>
</div>,
  "nodes": Array [
    <div>
      <h2>
            We found a few matches, is one of these what you are looking for?
      </h2>
      <div
            className="location-options">
            
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <LocationResults
        results={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LocationResults {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "results": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    We found a few matches, is one of these what you are looking for?
          </h2>
          <div
                    className="location-options">
                    
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
                    We found a few matches, is one of these what you are looking for?
          </h2>
          <div
                    className="location-options">
                    
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LocationResults
    results={Array []} />,
}
`;
